tests/ui/transmutes_expressible_as_ptr_casts.fixed

```diff
@@ -82,3 +82,9 @@ fn issue_10449() {
 
     let _x: u8 = unsafe { *(f as *const u8) };
 }
+
+// Pointers cannot be cast to integers in const contexts
+const fn issue_12402() {
+    unsafe { transmute::<*const i32, usize>(&42i32) };
+    unsafe { transmute::<fn(), usize>(issue_12402) };
+}
```

Comment:
I'd also add a scenario with the code that the user tried.

```suggestion
const fn issue_12402<P>(ptr: *const P) {
    unsafe { transmute::<*const i32, usize>(&42i32) };
    unsafe { transmute::<fn(), usize>(issue_12402) };
    let _ = unsafe { transmute::<_, usize>(ptr) };
}
```

Reply:
Done.